packages/flutter_tools/lib/src/macos/application_package.dart

```diff
@@ -170,6 +189,11 @@ class BuildableMacOSApp extends MacOSApp {
     final _BundleInfo bundleInfo = MacOSApp._executableFromBundle(globals.fs.directory(directory));
     return bundleInfo?.executable;
   }
+
+  static Future<BuildableMacOSApp> fromProject(MacOSProject project, BuildInfo buildInfo)async {
+    // todo projectBundleId need parse from buildInfo like IosProject
+    return BuildableMacOSApp(project, '');
```

Comment:
@jonahwilliams pass '' to fix assert error and leave a todo

Reply:
I will give another pr to fix this todo

Reply:
Generally I agree with @jonahwilliams 
> We don't use the id currently, so the current state of things is not actually an issue, but if you're going to open a PR to fix it I would prefer to see a completed fix instead of a temp hack.

See also our conversation from `BuildableIOSApp`:
https://github.com/flutter/flutter/pull/83956#discussion_r645771854

However it would actually add a few seconds to the macOS build to do this the "right way" and run a `xcodebuild -showBuildSettings` command to get the bundle ID.  You would have to implement a macOS version of https://github.com/flutter/flutter/blob/a69b3ca5d37ea0a9da71b24f72ae8584231a074b/packages/flutter_tools/lib/src/xcode_project.dart#L260
https://github.com/flutter/flutter/blob/a69b3ca5d37ea0a9da71b24f72ae8584231a074b/packages/flutter_tools/lib/src/xcode_project.dart#L300
https://github.com/flutter/flutter/blob/a69b3ca5d37ea0a9da71b24f72ae8584231a074b/packages/flutter_tools/lib/src/ios/xcodeproj.dart#L167
handle flavors, find and parse the Info.plist, etc.

Which seems like a lot since the `BuildableMacOSApp` id isn't used.  I'm actually okay with this staying a placeholder hack until it's actually used somewhere (maybe `com.example.placeholder` or something like that to make it obvious if it ever surfaces, rather than an empty string?), with a detailed comment that it's not currently used and so is a placeholder.

We'll need to work around this being null anyway when this file is migrated to null safety.  wdyt @jonahwilliams?